File: _a_arm/python_old/signalGenerator.py
import numpy as np

class signalGenerator:
    ''' 
        This class inherits the Signal class. It is used to organize 
        1 or more signals of different types: square_wave, 
        sawtooth_wave, triangle_wave, random_wave.
    '''
    def __init__(self, amplitude=1, frequency=1, y_offset=0):
        ''' 
            amplitude - signal amplitude.  Standard deviation for random.
            frequency - signal frequency
            y_offset  - signal y-offset
        '''
        self.amplitude = amplitude
        self.frequency = frequency
        self.y_offset = y_offset

    def square(self, t):
        if t % (1.0/self.frequency) <= 0.5/self.frequency:
            out = self.amplitude + self.y_offset
        else:
            out = - self.amplitude + self.y_offset
        return [out]
        #  returns a list of length 1

    def sawtooth(self, t):
        tmp = t % (0.5/self.frequency)
        out = 4*self.amplitude*self.frequency*tmp - self.amplitude + self.y_offset
        return [out]
        #  returns a list of length 1

    def random(self, t):
        out = np.sqrt(self.amplitude)*np.random.rand() + self.y_offset
        return [out]
        #  returns a list of length 1

    def sin(self, t):
        out = self.amplitude*np.sin(2*np.pi*self.frequency*t) + self.y_offset
        return [out]
        #  returns a list of length 1

File: _B_pendulum/matlab_old/signalGenerator.m
classdef signalGenerator
    %   produces: square_wave, sawtooth_wave, random_wave, sin_wave.
    %-----------------------------------
    properties
        amplitude
        frequency
        y_offset
    end
    %-----------------------------------
    methods
        %---constructor--------------------------------
        function self = signalGenerator(amplitude, frequency, y_offset)
            self.amplitude = amplitude;
            if nargin>1
                self.frequency = frequency;
            else
                self.frequency = 1;
            end
            if nargin>2
                self.y_offset = y_offset;
            else
                self.y_offset = 0;
            end
        end
        %-----------------------------------
        function out = square(self, t)
            if mod(t, 1/self.frequency)<=0.5/self.frequency
                out = self.amplitude + self.y_offset;
            else
                out = -self.amplitude + self.y_offset;
            end
        end
        %-----------------------------------
        function out = sawtooth(self, t)
            out = 4*self.amplitude*self.frequency * mod(t, 0.5/self.frequency)...
                - self.amplitude + self.y_offset;
        end
        %-----------------------------------
        function out = random(self, t)
            out = sqrt(self.amplitude)*randn + self.y_offset;
        end
        %-----------------------------------
        function out = sin(self, t)
            out = self.amplitude*sin(2*pi*self.frequency*t) + self.y_offset;
        end
    end
end